pkgs/applications/editors/jetbrains/darwin.nix

```diff
@@ -27,10 +27,10 @@ stdenvNoCC.mkDerivation {
     runHook preInstall
     APP_DIR="$out/Applications/${product}.app"
     mkdir -p "$APP_DIR"
-    cp -Tr "${product}.app" "$APP_DIR"
+    cp -Tr *.app "$APP_DIR"
     mkdir -p "$out/bin"
     cat << EOF > "$out/bin/${loname}"
-    open -na '$APP_DIR' --args "\$@"
+    open -na '$APP_DIR/Contents/MacOS/${loname}' --args "\$@"
```

Comment:
Opening `$APP_DIR` looks much more general. Is there a reason to do this instead?

Reply:
I copied it directly from the script that JetBrains Toolbox generates. Perhaps we want to generate that whole script here?

```bash
#!/bin/bash
#Generated by JetBrains Toolbox 2.0.4.17212 at 2023-10-02T19:38:26.685983

declare -a intellij_args=()
declare -- wait=""

for o in "$@"; do
  if [[ "$o" = "--wait" || "$o" = "-w" ]]; then
    wait="-W"
    o="--wait"
  fi
  if [[ "$o" =~ " " ]]; then
    intellij_args+=("\"$o\"")
  else
    intellij_args+=("$o")
  fi
done

open -na "$APP_DIR/Contents/MacOS/rustrover" $wait --args "${intellij_args[@]}"
```

Reply:
I'm wondering if `$APP_DIR/Contents/MacOS/${loname}` is valid for all JetBrains IDEs. Have you run a check via `nixpkgs-review pr 258814`?

Reply:
I went ahead and reverted back to `$APP_DIR`, since [this doc](https://www.jetbrains.com/help/rust/working-with-the-ide-features-from-command-line.html) says is OK. I think all we lose from the `--wait` flag, described as follows: 

> Wait for the files to be closed before returning to the command prompt.
> 
> For example, you can open file.txt with the following command:
>
> rust --wait file.txt
>
> The shell will be waiting until file.txt is closed.

I'm not sure how important that is, but I would consider it an enhancement to the `jetbrains.*`, as opposed to just getting the `EAP` to work. I may consider that in a separate PR.